packages/aws-arch/README.md

```diff
@@ -0,0 +1,90 @@
+# aws-arch (AWS Architecture)
+
+Library to provide metadata for AWS Services and AWS Resources.
+
+This package generates mappings between [@aws-cdk/cfnspec](https://github.com/aws/aws-cdk/blob/main/packages/%40aws-cdk/cfnspec) and [AWS Architecture Icons](https://aws.amazon.com/architecture/icons/), and resolves resource metadata between these systems to infer a normalized definition of an `AwsService` and `AwsResource`.
+
+The primary aim of this package is to provide a consistent mechanism for other libraries to retrieve naming and assets associated with CloudFormation resources, and by extension CDK resources.
+
+## Get Started
+
+```sh
+yarn add '@aws-prototyping-skd/aws-arch'
+```
+
+```ts
+import { AwsArchitecture } from "@aws-prototyping-skd/aws-arch";
+
+const s3Bucket = AwsArchitecture.getResource("AWS::S3::Bucket");
+const s3Service = AwsArchitecture.getService(s3Bucket.service);
+
+console.log(s3Bucket);
+console.log(s3Service);
+```
+
+```js
+// => console.log(s3Bucket);
+{
+	"name": "Amazon Simple Storage Service Bucket",
+	"cfnType": "AWS::S3::Bucket",
+	"awsAssetName": "Amazon-Simple-Storage-Service_Bucket",
+	"awsAssetIcon": "resources/Amazon-Simple-Storage-Service_Bucket.png",
+	"service": "S3"
+}
+
+// => console.log(s3Service);
+{
+	"provider": "AWS",
+	"name": "Amazon Simple Storage Service",
+	"cfnName": "S3",
+	"awsAssetIcon": "services/Amazon-Simple-Storage-Service.png",
+	"awsAssetName": "Amazon-Simple-Storage-Service"
+}
+```
+
+## Aws Achritecture Icons
+
+Retrieve **category**, **service**, and **resource** [AWS Architecture Icons](https://aws.amazon.com/architecture/icons/).
+
+> Icon methods return relative asset key paths, as most frameworks have the concept of a base path (imagepaths). Use `AwsArchitecture.resolveAssetPath(...)` to get absolute path.
+
+### Retrieve icon based on CloudFormation Resource Type
+
+**Resource Icon**
+
+```ts
+const s3Bucket = AwsArchitecture.getResource("AWS::S3::Bucket");
+
+const s3BucketPng = s3Bucket.icon("png"); // => "storage/s3/bucket.png"
```

Comment:
should the image types be typed i.e. enumified?

Reply:
They are typed as a string literal union. The reason being is the jsii enums don't allow literal value comparison, and wanted to not require importing an enum for this.

https://aws.github.io/jsii/specification/2-type-system/#enum

Reply:
Discussed offline - resolving